Added display_crawl_results

vringar opened the pull request:
This function should give the user some general overview over the crawl_history and what kind of data loss to expect.

vringar committed:
Added mode parameter to PySparkS3Dataset

This parameter allows for filtering out VisitIds that are part of
`incompleted_visits` or that had a command with a command_status other than
"ok" since users probably shouldn't consider them for analysis

This filtering functionality is extracted into the TableFilter class to
be reused by other Datasets.

vringar committed:
Added display_crawl_results

vringar committed:
Rewrote crawlhistory.py

englehardt commented on openwpm_utils/crawlhistory.py:
```diff
@@ -46,3 +48,127 @@ def get_worst_status_per_visit_id(crawl_history):
 
+def display_crawl_history_per_command_sequence(crawl_history, interrupted_visits):
+    """
+        Analyze crawl_history and interrupted_visits to display general
```
please use numpy style source docs for these functions.

englehardt commented:
@vringar do you have a sample notebook I can use to check the outputs of these? I know we don't have tests (which is okay for now), but that would help me check whether everything is what I'd expect.

englehardt reviewed (changes requested):
.

vringar commented:
> @vringar do you have a sample notebook I can use to check the outputs of these? I know we don't have tests (which is okay for now), but that would help me check whether everything is what I'd expect.

This function is used in the [dataquality notebook](https://dbc-caf9527b-e073.cloud.databricks.com/#notebook/235123/command/294089) on Databricks.

vringar committed:
Used typeannotations

Commit:
Fixing display_crawl_history

Commit:
Added docstrings

Commit:
Added demo file

Commit:
Backporting from next

Commit:
Removed collect_content from PySparkS3Dataset

Downloading files via the SparkContext was much slower than
downloading via boto (which is what S3Dataset does.
So now both classes use the same method, as PySparkS3Dataset
inherits from S3Dataset
